leaderboard/enriching/enricher.go

```diff
@@ -0,0 +1,128 @@
+package enriching
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	podium_leaderboard_webhooks_v1 "github.com/topfreegames/podium/leaderboard/v2/enriching/proto/webhook/v1"
+	"github.com/topfreegames/podium/leaderboard/v2/model"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+const enrichURL = "/leaderboards/enrich"
+
+type EnrichmentConfig struct {
+	// WebhookUrls contains the necessary parameters to call a webhook for a given game.
+	// The key should be the game tenantID.
+	WebhookUrls map[string]string `mapstructure:"webhook_urls"`
+}
+
+type enricherImpl struct {
+	config EnrichmentConfig
+}
+
+// NewEnricher returns a new Enricher implementation.
+func NewEnricher(config EnrichmentConfig) Enricher {
+	return &enricherImpl{
+		config: config,
+	}
+}
+
+func (e *enricherImpl) Enrich(tenantID, leaderboardID string, members []*model.Member) ([]*model.Member, error) {
+	tenantUrl, exists := e.config.WebhookUrls[tenantID]
+	if !exists {
+		return members, nil
+	}
+
+	body := membersModelToProto(leaderboardID, members)
+	jsonData, err := json.Marshal(body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to call webhook: %w", errors.Join(err, ErrEnrichmentInternal))
```

Comment:
I think having all error messages return basically the same message (except for the `err`) will make our troubleshooting hard, shouldn't we at least log the operation that we are trying to do? ex: marshalling, building url, etc.

Reply:
Updated error messages. The loggin will happen on the routes where enrich is called